packages/codegen/src/with-cache.ts

```diff
@@ -37,9 +38,10 @@ export const withCache =
     stack.delete(id)
 
     if (cache.has(id)) {
-      result = onExitCircular(result, cache.get(id)!, input, ...rest)
+      return onExitCircular(result, cache.get(id)!, input, ...rest)
```

Comment:
so the idea is that `onExitCircular` can alter what's in cache, and the cache would need to be overwritten?

Reply:
Yes - `withCache` is a caching function that handles cricularities. Imagine this case:

`onEnterCircular` is meant to provide a temporary value to close the circularity, and `onExitCircular` is meant to replace the value once the other part is finished.

In this case it's not being used though, because with the regular type codegen circularities are broken with anonymous types (so we just need the checksum which we can know onEnterCircular) and remain unchanged onExit, and for docs codegen you _could_ link the circular node back to itself onExit, but then it would blow up when stringifying it into a type.

Reply:
okay, rolled back that part